### ISSUE TITLE
Are there any conditions under which the logs of the Entitlement Analyzer are deleted?

### ISSUE THREAD

**Issue by skozawa-kka** [qlik-oss/qlik-cloud-entitlement-analyzer#37]
I hope this message finds you well.

I understand that the Entitlement Analyzer retrieves logs via the REST Connector.

Regarding these logs, are there any conditions under which the logs are deleted, 
such as when the logs reach a certain file size or after a certain number of days have passed? 

If there are such conditions, could you please provide the specific conditions?

Our customer wants to check the logs from several years ago, so I am inquiring about this.

Thank you for your assistance.

Best regards,

Shiho Kozawa

**Comment by danielpilla** [comment]
Hi @skozawa-kka ,
If you are referring to "logs" as the usage data available via API from `api/v1/licenses/consumption`, that data from that API does indeed appear to go back for several years, however the Entitlement Analyzer does not support fetching data from that far back. It pulls from the current month and then builds forward indefinitely. The [App Analyzer](https://github.com/qlik-oss/qlik-cloud-app-analyzer), however, which should now be used instead for this type of data analysis as data from this API has been deprecated in the Entitlement Analyzer in its most recent version, begins by going back 90 days and then building forward. The App Analyzer does however have a variable that you can set for the total duration stored within the QVDs, where the default is 365 days rolling. You can increase this to store more data over time, but it will begin with 90 days worth of data.

`Set vu_session_rolling_range 			= 365; 	// Maximum days back to store session data in QVDs -- default 365`

The App Analyzer does not support going back beyond 90 days as well due to other APIs having limitations of 90 days, therefore the data could not be joined.

In summary, you could fetch deeper historical data from the `api/v1/licenses/consumption` API manually through your own mechanism but not through any of the monitoring apps.

Cheers,

**Comment by skozawa-kka** [comment]
Hi @danielpilla 
Thank you for your response.

Your explanation helped me gain a deeper understanding of Entitlement Analyzer and App Analyzer.

Regarding the logs via API from api/v1/licenses/consumption, is the following understanding correct?

・It is possible to go back to the point when the customer started using the SaaS environment.
・The logs remain indefinitely.

Best regards,

Shiho Kozawa

**Comment by danielpilla** [comment]
Hi @skozawa-kka ,
In theory, it should be, yes, but it is not documented and not supported. I tested on a tenant that I have had since 2019 and my data goes back to May 2021 for that API endpoint. The standard for many of our APIs is 90 days, so I might not expect that to always be the case as they could change it at some point. That said, there are other archival endpoints like [audits/archive](https://qlik.dev/apis/rest/audits/#get-v1-audits-archive) that will store events for the history of the tenant. It might not have the data points you seek, however.

Cheers,

**Comment by skozawa-kka** [comment]
Hi @danielpilla 

Thank you very much.
I have gained a deeper understanding of the logs.

Best regards,
